### PR TITLE
Vibe -- fix undefined behavior, gargabe output

### PR DESCRIPTION
**Pull request by x42**
Ideally apply on top of #36 

This fixes various issues with rkr-vibe due to uninitialized variables in the filters.
Depending on the initial state, it may produce constant garbage output and even NaN or Inf.

### CODE REVIEW

**Comment by x42** [comment]
PS. Other effects may have similar issue, but before updating those, there'd ideally be a consistent API to `::activate()` a plugin. That should called only once when leaving bypass to clear and reset stat,  separate from `::cleanup()`.

File: src/Vibe.C [hunk 1]
```diff
@@ -311,7 +319,7 @@ Vibe::vibefilter(float data, fparams *ftype, int stage)
 };
 
 float
-Vibe::bjt_shape(float data)
+Vibe::bjt_shape(float data) const
```
Comment (ssj71) [inline src/Vibe.C:322]: for my own education, is this const making it return a const (immutable) float? what is the motivation for this change?

**Comment by ssj71** [comment]
so you are saying ::cleanup should not be called when bypassing, and instead there should be a proper ::activate? I think I just implemented the same logic as was in the standalone rakarrack code to call cleanup on bypass. Doesn't make it right, I'm just trying to understand.

I'm fine with merging this, but I'd like to understand it a bit better.

File: src/Vibe.C (hunk 1 above)
Comment (x42) [reply src/Vibe.C:322]: it means that a method does not modify any class member variables.

I've added this while debugging, to ensure that `vibefilter()` and `bjt_shape()` don't change the state. Both methods could actually be static, since they don't use any member vars.

**Comment by x42** [comment]
> so you are saying ::cleanup should not be called when bypassing.

Ideally it would called only once.  Right now every run_*() call periodically calls the cleanup method when a  plugin is bypassed `if(*plug->bypass_p && plug->prev_bypass)` is true after the x-fade completed.

**Comment by x42** [comment]
The main part of the fix is the `fparams()` constructor in  src/Vibe.h that initializes `x1 = y1 = 0;`
Previously those were uninitialized variables, producing garbage audio output.

After a bypass/re-enable those variable should be cleared, as well. The current patch clears them every cycle. It's not expensive to do that, so it's fine for the time being.

> so you are saying ::cleanup should not be called when bypassing, and instead there should be a proper ::activate? 

Yes.   Long term it would be nice to only reset the state once.

**Comment by x42** [comment]
context: https://discourse.ardour.org/t/all-of-the-lv2-vibe-plugins-are-broken-underlying-issue/101628

**Comment by ssj71** [comment]
ok, I think I should go through and fix the logic to only call ::cleanup  excatly once after the xfade out. That was the intended result but I can see that its not correct. I can hopefully find enough time to do that in the next few days, that shouldn't necessarily change this PR though IIUC. Sound alright?

**Comment by x42** [comment]
Ideally it'd be called once before fade_in: When activating a plugin, including the first run.

There's usually no need to reset state during cleanup.

**Comment by ssj71** [comment]
I'm thinking like #38 

It should get the job done, only that it will run an uncessary call to cleanup() when the plugin is bypassed then removed. Changing it to run the other way (cleanup on activate) is just a bit more work that I'm not sure is worth it. Am I missing something?
